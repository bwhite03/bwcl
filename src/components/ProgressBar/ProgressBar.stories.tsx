import React from 'react';
import { Story, Meta } from '@storybook/react';
import ProgressBar, { ProgressBarProps } from './ProgressBar';

export default {
  title: 'Progress Bar',
  component: ProgressBar,
} as Meta;

const Template: Story<ProgressBarProps> = (args) => <ProgressBar {...args} />;
export const ProgressBarExample = Template.bind({});
ProgressBarExample.args = {
  msg: 'Progress',
  max: 100,
  progress: 50,
};
