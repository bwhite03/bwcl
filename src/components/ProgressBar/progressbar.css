.bwcl-progress-bar {
  height: 30px;
  width: 300px;
  background-color: #d8d8d8;
  border-radius: 20px;
  margin-bottom: 5px;
  position: absolute;
  left: 200;
  top: 20;
  opacity: 0.8;
  box-shadow: 2px 4px 10px rgba(0, 0, 0, 0.3);
}

.bwcl-progress-done {
  background: linear-gradient(to left, #3d61d9, #8062c4);
  box-shadow: 0 3px 3px -5px #3d61d9, 0, 2px 5px #3d61d9;
  color: #fff;
  border-radius: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  transition: 1s ease;
}

.bwcl-progress-msg1 {
  color: #fff;
  transform: translateY(-135%) translateX(100px);
}
