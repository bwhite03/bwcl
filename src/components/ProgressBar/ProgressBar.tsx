import React, { useEffect, useState } from 'react';
import './progressbar.css';

export interface ProgressBarProps {
  msg: string;
  max: number;
  progress: number;
  opacity?: number;
  msgStyle?: React.CSSProperties;
}

const ProgressBar = (props: ProgressBarProps) => {
  const [style, setStyle] = useState<React.CSSProperties>({});
  const [progress, setProgress] = useState(0);
  const {
    msgStyle = {
      color: 'rgb(22, 10, 90)',
      transform: 'translateY(-135%) translateX(100px)',
    },
  } = props;

  useEffect(() => {
    const currentProgress = (props.progress / props.max) * 100;
    setProgress(currentProgress);
    const newStyle = {
      width: `${currentProgress}%`,
    };

    const newMsgStyle = {
      ...props.msgStyle,
    };

    setStyle(newStyle);
  }, [props.progress]);

  return (
    <div className="bwcl-progress-bar">
      <div className="bwcl-progress-done" style={style} />
      <div
        className="bwcl-progress-msg"
        style={msgStyle}
      >{`${props.msg} ${progress}`}</div>
    </div>
  );
};

export default ProgressBar;
