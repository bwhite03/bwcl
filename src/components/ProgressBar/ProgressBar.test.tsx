import React from 'react';
import { render } from '@testing-library/react';
import ProgressBar from './ProgressBar';

describe('ProgressBar', () => {
  test('should render correctly', () => {
    const { container } = render(
      <ProgressBar msg="progress" max={100} progress={50} />
    );

    expect(container).toMatchSnapshot();
  });
});
