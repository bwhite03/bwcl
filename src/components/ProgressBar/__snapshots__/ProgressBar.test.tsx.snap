// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render correctly 1`] = `
<div>
  <div
    class="bwcl-progress-bar"
  >
    <div
      class="bwcl-progress-done"
      style="width: 50%;"
    />
    <div
      class="bwcl-progress-msg"
      style="color: rgb(22, 10, 90); transform: translateY(-135%) translateX(100px);"
    >
      progress 50
    </div>
  </div>
</div>
`;
